packages/malloy/src/model/malloy_query.ts

```diff
@@ -593,77 +775,143 @@ class QueryField extends QueryNode {
-            // ) +
-            ` ${ordering.dir || 'ASC'}`
-        );
-      } else if (resultStruct.firstSegment.type === 'project') {
-        obSQL.push(
-          ` ${orderingField.fif.f.generateExpression(resultStruct)} ${
-            ordering.dir || 'ASC'
-          }`
-        );
+      if (fieldsString.length > 0) {
+        partitionBy += `, ${fieldsString}`;
       }
     }
-    const func = malloyFunctions[expr.function];
-    const paramSQL: string[] = [];
-    if (expr.parameters !== undefined) {
-      for (const e of expr.parameters) {
-        if (typeof e === 'string') {
-          paramSQL.push(e); // need to map to dimensional expression.
-        } else if (typeof e === 'number') {
-          paramSQL.push(e.toString());
+
+    // TODO: so order by and arguments need to generate partitionSQL if
+    // the partitionSQL exists and the parameter is an "output"
+
+    let orderBy = '';
+    if (overload.needsWindowOrderBy) {
+      // calculate the ordering.
+      const obSQL: string[] = [];
+      let orderingField;
+      const orderByDef =
+        (resultStruct.firstSegment as QuerySegment).orderBy ||
+        resultStruct.calculateDefaultOrderBy();
+      for (const ordering of orderByDef) {
+        if (typeof ordering.field === 'string') {
+          orderingField = {
+            name: ordering.field,
+            fif: resultStruct.getField(ordering.field),
+          };
         } else {
-          paramSQL.push(
-            this.generateExpressionFromExpr(resultStruct, context, e, state)
+          orderingField = resultStruct.getFieldByNumber(ordering.field);
+        }
+        const exprType = orderingField.fif.f.fieldDef.expressionType;
+        // TODO today we do not support ordering by analytic functions at all, so this works
+        // but eventually we will, and this check will just want to ensure that the order field
+        // isn't the same as the field we're currently compiling (otherwise we will loop infintely)
+        if (expressionIsAnalytic(exprType)) {
+          continue;
+        }
+        if (resultStruct.firstSegment.type === 'reduce') {
+          let orderSQL = orderingField.fif.getPartitionSQL();
+          if (
+            context.dialect.requiresWindowOrderByToBeGrouped &&
+            expressionIsScalar(exprType)
+          ) {
+            // ANY_VALUE here is a no-op which turns the expression into an aggregate, which allows
+            // BigQuery to use a field which is not in the group_by directly (this happens when the
+            // order_by of the window function (from the query's order_by) is on a field which is an
+            // expression rather than a column -- therefore the group_by contains an expression based
+            // on a field and not the field itself, causing an error when it's referenced in the window
+            // order_by.
+            // TODO decide if there's a better way to do this
+            orderSQL = `ANY_VALUE(${orderSQL})`;
```

Comment:
Lloyd might have an opinion here. Carlin gave me this idea to fix an issue where expressions involving fields that are in the output don't work in the order by in a window function's OVER clause. So in BigQuery, for these we wrap them in `ANY_VALUE` to turn them into aggregates, which are allowed, while not actually changing the semantics.